frontend/packages/console-dynamic-plugin-sdk/src/api/component-api-types.ts

```diff
@@ -0,0 +1,16 @@
+import { RouteComponentProps, match } from 'react-router';
+import { K8sResourceCommon } from '../extensions';
+
+/* Horizontal Nav Types */
+export type NavPage = {
+  href?: string;
+  path?: string;
+  name?: string;
+  component?: React.ComponentType<RouteComponentProps>;
+};
```

Comment:
`name` is required no?
Same with `component`?

Reply:
You're right. Initially it was optional so I kept as it is. But can't think of any use cases.